tests/unit/components/panel/AttributeSection.spec.js

```diff
@@ -0,0 +1,325 @@
+            name: 'test',
+            type: 'String',
+          }],
+        },
+      })).toEqual([new ComponentAttribute({
+        name: 'test',
+        type: 'String',
+        definition: {
+          name: 'test',
+          type: 'String',
+        },
+      })]);
+    });
+
+    it('should return instantiate defined attribute with its value even if value is not empty', () => {
+      expect(wrapper.vm.getSubAttributes({
+        value: [{
+          name: 'test',
+          value: 'toto',
+          type: 'String',
+          definition: {
+            name: 'test',
+            type: 'String',
+          },
+        }],
+        definition: {
+          definedAttributes: [{
+            name: 'test',
+            type: 'String',
+          }],
+        },
+      })).toEqual([new ComponentAttribute({
+        name: 'test',
+        type: 'String',
+        value: 'toto',
+        definition: {
+          name: 'test',
+          type: 'String',
+        },
+      })]);
+    });
+
+    it('should return instantiate attribute with its value even if definition not exists', () => {
+      expect(wrapper.vm.getSubAttributes({
+        value: [{
+          name: 'test',
+          value: 'toto',
+          type: 'String',
+        }],
+      })).toEqual([{
+        name: 'test',
+        type: 'String',
+        value: 'toto',
+      }]);
+    });
+  });
+
+  describe('Test function: addAttribute', () => {
+    it('should emit event', async () => {
+      await wrapper.vm.addAttribute();
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'attribute',
+        attribute: {
+          name: 'attribute',
+          type: 'String',
+          value: [{
+            name: 'attribute_1',
+            type: 'String',
+            value: '',
+            definition: null,
+          }],
+        },
+      }]]);
+    });
+  });
+
+  describe('Test function: deleteAttribute', () => {
+    it('should emit event', async () => {
+      await wrapper.vm.deleteAttribute();
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'attribute',
+      }]]);
+    });
+  });
+
+  describe('Test function: updateAttributeName', () => {
+    it('should emit event', async () => {
+      await wrapper.vm.updateAttributeName('test');
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'test',
+        attribute: {
+          name: 'test',
+          type: 'String',
+          value: [],
+        },
+      }]]);
+    });
+  });
+
+  describe('Test function: updateAttributeValue', () => {
+    it('should emit event', async () => {
+      await wrapper.vm.updateAttributeValue('test');
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'attribute',
+        attribute: {
+          name: 'attribute',
+          type: 'String',
+          value: 'test',
+        },
+      }]]);
+    });
+  });
+
+  describe('Test function: updateAttribute', () => {
```

Comment:
1. beforeEach with setProps
2. use var/const to avoid duplications

Reply:
done